chainextensions/dapps-staking/impl/src/lib.rs

```diff
@@ -0,0 +1,489 @@
+                    .ok_or(DispatchError::Other("general_era_info call failed"));
+                let staked_amount = era_info.map_or(Zero::zero(), |r| r.staked);
+                env.write(&staked_amount.encode(), false, None)
+                    .map_err(|_| {
+                        DispatchError::Other(
+                            "[ChainExtension] DappsStakingExtension failed to write result",
+                        )
+                    })?;
+            }
+
+            // DappsStaking - read_staked_amount()
+            DappsStakingFunc::StakedAmount => {
+                let staker: T::AccountId = env.read_as()?;
+                let ledger = pallet_dapps_staking::Ledger::<T>::get(&staker);
+                env.write(&ledger.locked.encode(), false, None)
+                    .map_err(|_| {
+                        DispatchError::Other(
+                            "[ChainExtension] DappsStakingExtension failed to write result",
+                        )
+                    })?;
+            }
+
+            DappsStakingFunc::StakedAmountOnContract => {
+                let args: DappsStakingAccountInput = env.read_as()?;
+                let staker = T::AccountId::decode(&mut args.staker.as_ref()).unwrap();
+                let contract = Self::decode_smart_contract(args.contract)?;
+                let staking_info =
+                    pallet_dapps_staking::GeneralStakerInfo::<T>::get(&staker, &contract);
+                let staked_amount = staking_info.latest_staked_value();
+                env.write(&staked_amount.encode(), false, None)
+                    .map_err(|_| {
+                        DispatchError::Other(
+                            "[ChainExtension] DappsStakingExtension failed to write result",
+                        )
+                    })?;
+            }
+
+            // DappsStaking - read_contract_stake()
+            DappsStakingFunc::ReadContractStake => {
+                let contract_bytes: [u8; 32] = env.read_as()?;
+                let contract = Self::decode_smart_contract(contract_bytes)?;
+                let current_era = pallet_dapps_staking::CurrentEra::<T>::get();
+                let staking_info =
+                    pallet_dapps_staking::Pallet::<T>::contract_stake_info(&contract, current_era)
+                        .unwrap_or_default();
+                let total = TryInto::<u128>::try_into(staking_info.total).unwrap_or(0);
+                env.write(&total.encode(), false, None).map_err(|_| {
+                    DispatchError::Other(
+                        "[ChainExtension] DappsStakingExtension failed to write result",
+                    )
+                })?;
+            }
+
+            // DappsStaking - register()
+            DappsStakingFunc::Register => {
+                let contract_bytes: [u8; 32] = env.read_as()?;
+                let contract = Self::decode_smart_contract(contract_bytes)?;
```

Comment:
If this isn't updated to support EVM (and other contracts), you should consider implementing `From<[u8; 32]>` for the smart contract type and just use `into()`.

Reply:
I'll fix it in when Implementing staking for EVM contract as well